src/main/scala/com/amazon/deequ/comparison/DataSynchronization.scala

```diff
@@ -0,0 +1,131 @@
+/** Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+  *
+  * Licensed under the Apache License, Version 2.0 (the "License"). You may not
+  * use this file except in compliance with the License. A copy of the License
+  * is located at
+  *
+  * http://aws.amazon.com/apache2.0/
+  *
+  * or in the "license" file accompanying this file. This file is distributed on
+  * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+  * express or implied. See the License for the specific language governing
+  * permissions and limitations under the License.
+  */
+
+package com.amazon.deequ.comparison
+
+import org.apache.spark.sql.{Column, DataFrame}
+import org.apache.spark.sql.functions.col
+
+object DataSynchronization {
+
+  /** @param df1:
+    *   The dataset from which the keys of the mergedMap are retrieved from.
+    * @param df2:
+    *   The dataset from which the values of the mergedMap are retrieved from.
+    * @param mergedMaps:
+    *   A map that contains the columns that the user is going to compare.
+    * @return
+    *   Double: Returns the amount of rows of the joined DataFrame as a Double
+    */
+  def mergingMaps(
+      df1: DataFrame,
+      df2: DataFrame,
+      mergedMaps: Map[String, String]
+  ): Double = {
+    val joinExpression: Column = mergedMaps
+      .map { case (col1, col2) =>
+        df1(col1) === df2(col2)
+      }
+      .reduce((e1, e2) => e1 && e2)
+
+    // joining the df1 with df2 with only the rows that matches obtained from joinExpression
+    val joined = df1.join(df2, joinExpression, "inner")
```

Comment:
see below, I suspect doing a normal join and then figuring out how many rows were lost is faster

Reply:
maybe even using a semi-join + caching